tests/commands/test_api_error.py

```diff
@@ -0,0 +1,112 @@
+import os
+import tempfile
+from unittest import mock
+from pathlib import Path
+import responses
+from launchable.utils.http_client import get_base_url
+from tests.cli_test_case import CliTestCase
+
+
+class APIErrorTest(CliTestCase):
+    test_files_dir = Path(__file__).parent.joinpath(
+        '../data/minitest/').resolve()
+
+    @responses.activate
+    @mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
+    def test_record_build(self):
+        responses.add(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/builds".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace), status=500)
+
+        result = self.cli("record", "build", "--name", "example")
+        self.assertEqual(result.exit_code, 0)
+
+    @responses.activate
+    @mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
+    def test_record_session(self):
+        build = "internal_server_error"
+        responses.add(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/builds/{build}/test_sessions".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace, build=build), status=500)
+
+        result = self.cli("record", "session", "--build", build)
+        self.assertEqual(result.exit_code, 0)
+
+        build = "not_found"
+        responses.add(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/builds/{build}/test_sessions".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace, build=build), status=404)
+
+        result = self.cli("record", "session", "--build", build)
+        self.assertEqual(result.exit_code, 1)
+
+    @responses.activate
+    @mock.patch.dict(os.environ, {"LAUNCHABLE_TOKEN": CliTestCase.launchable_token})
+    def test_subset(self):
+        responses.replace(responses.POST, "{base}/intake/organizations/{org}/workspaces/{ws}/subset".format(
+            base=get_base_url(), org=self.organization, ws=self.workspace), status=500)
+
+        subset_file = "example_test.rb"
+
+        rest_file_500 = tempfile.NamedTemporaryFile(delete=False)
+        result = self.cli("subset", "--target", "30%", "--session", self.session, "--rest", rest_file_500.name,
+                          "minitest", str(self.test_files_dir) + "/test/**/*.rb", mix_stderr=False)
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual(len(result.stdout.rstrip().split("\n")), 1)
+        self.assertTrue(subset_file in result.stdout)
+
+        rest = Path(rest_file_500.name).read_text()
+        self.assertEqual(
+            len(rest.rstrip().split("\n")), 1)
+        self.assertTrue(subset_file in rest)
+
+        # to avoid permission denied error on Windows
+        rest_file_500.close()
+        os.unlink(rest_file_500.name)
```

Comment:
[nits]
this can be written with `with` clause to avoid forgetting to delete the named temp file.

```python
        with tempfile.NamedTemporaryFile() as rest_file_500:
                result = self.cli("subset", "--target", "30%", "--session", self.session, "--rest", rest_file_500.name,
                          "minitest", str(self.test_files_dir) + "/test/**/*.rb", mix_stderr=False)

                self.assertEqual(result.exit_code, 0)
                self.assertEqual(len(result.stdout.rstrip().split("\n")), 1)
                self.assertTrue(subset_file in result.stdout)

                rest = Path(rest_file_500.name).read_text()
                self.assertEqual(len(rest.rstrip().split("\n")), 1)
                self.assertTrue(subset_file in rest)
```

Reply:
I worried about does it succeed in Windows environment, but it passed.
So, I'll fix other codes https://github.com/launchableinc/cli/runs/5441841695?check_suite_focus=true